controllers/users-controller.js

```diff
@@ -156,3 +168,51 @@ exports.logoutUser = (appReq, appRes) => {
     token: null,
   });
 };
+
+
+/**
+ * @description Handles resetting password
+ *
+ * @api {POST} /users/reset-password
+ * @apiSuccess 200 {auth: true, token: token} jsonwebtoken.
+ * @apiError 401 {auth: false, token: null} Invalid password.
+ * @apiError 404 {request error} User not found.
+ * @apiError 500 {server error} Problem finding user.
+ *
+ * @param {string} appReq.body.email - email provided by user
+ * @param {string} appReq.body.password - user provided password
+ * @param {string} appReq.body.new_password - user provided password
+ * @param {string} appReq.body.confirm_password - user provided password
+ */
+exports.resetPassword = (appReq, appRes) => {
+
+  const HASHED_PASSWORD = bcrypt.hashSync(appReq.body.new_password, 8);
+
+  User.findOneAndUpdate({ email: appReq.body.email }, {password: HASHED_PASSWORD}, {upsert:false}, (err, user) => {
+    if (err) return appRes.status(500).send('Error on the server.');
+    if (!user) return appRes.status(404).send('No user found.');
+
+    const passwordIsValid = bcrypt.compareSync(
```

Comment:
maybe we can find a way to declare once and make available to both login & reset